manifests/master/generate_types.pp

```diff
@@ -99,56 +115,100 @@
     }
   }
 
-  if $trigger_on_puppetserver_update {
-    incron::system_table { 'simp_generate_types_puppetserver_exe':
-      enable         => $enable,
-      custom_content => epp(
-        "${module_name}/simp_generate_types_incron_rules/puppetserver.epp",
-        {
-          'simp_generate_types' => $_generate_types_path,
-          'puppetserver_exe'    => $puppetserver_exe,
-          'run_dir'             => $run_dir
-        }
-      ),
-      require        => [
-        File[$run_dir],
-        File[$_generate_types_path]
-      ]
+  if ($monitor_type == 'systemd') and ('systemd' in pick(fact('init_systems'), [])) {
+    simplib::assert_optional_dependency($module_name, 'camptocamp/systemd')
+
+    systemd::unit_file { 'simp_generate_types.path':
+      enable  => true,
+      active  => true,
+      content => epp("${module_name}/etc/systemd/system/simp_generate_types.path.epp")
     }
-  }
-  else {
-    incron::system_table { 'simp_generate_types_puppetserver_exe': enable => false }
-  }
 
-  if $trigger_on_puppet_update {
-    incron::system_table { 'simp_generate_types_puppet_exe':
-      enable         => $enable,
-      custom_content => epp(
-        "${module_name}/simp_generate_types_incron_rules/puppet.epp",
-        {
-          'simp_generate_types' => $_generate_types_path,
-          'puppet_exe'          => $puppet_exe,
-          'run_dir'             => $run_dir
-        }
-      ),
-      require        => [
-        File[$run_dir],
-        File[$_generate_types_path]
-      ]
+    $simp_generate_types_service = @("HEREDOC")
+      [Service]
+      Type=simple
+      ExecStart=${_generate_types_path} -d ${delay} -s -p ALL
+      | HEREDOC
+
+    systemd::unit_file { 'simp_generate_types.service':
+      content => $simp_generate_types_service
```

Comment:
Does this result in a `service` resource under the hoods, or should `svckill` be notified to ignore this service.

Reply:
There's a service right below it.